tests/dags/maintenance/test_rotate_db_snapshots.py

```diff
@@ -0,0 +1,48 @@
+from unittest import mock
+
+import boto3
+import pytest
+from maintenance.rotate_db_snapshots import delete_previous_snapshots
+
+
+@pytest.fixture
+def rds_client(monkeypatch):
+    rds = mock.MagicMock()
+
+    def get_client(*args, **kwargs):
+        return rds
+
+    monkeypatch.setattr(boto3, "client", get_client)
+    return rds
+
+
+def _make_snapshot(_id):
+    return {"DBSnapshotIdentifier": _id}
+
+
+@pytest.mark.parametrize(
```

Comment:
Just a note, if we also allow the snapshots to be configured via function argument as I describe in my comment above, we could try out values other than `7` here to make sure those work in the tests too 😄